### ISSUE TITLE
New Mode Suggestion: Recently View Pages

### ISSUE THREAD

**Issue by eejdoowad** [lusakasa/saka#27]
I often find myself unable to locate a page I've seen recently  either because I've navigated away from it or because I've closed it.

I propose a mode that presents a list of pages in order of view recency. This list should be updated when the user navigates to a new page within a tab or when the active tab changes.

When the user activate an entry in this mode, one of the following happens:
1) if page exists in some tab: switch to it
2) if page exists in a closed tab session: restore the session
3) otherwise: create a new tab with the page

Any thoughts?

**Comment by pureooze** [comment]
So this would be a combination of history, tabs and recently closed mode, sorted by last visited? I can see why that would be useful if you just want to do a search across the three modes without going to each one separately. Looking through the webextension docs, it looks like we can use the following:

- **Tabs**: lastAccessed ([here](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/tabs/Tab))
- **Sessions**: lastModified ([here](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/sessions/Session))
- **History**: lastVisitTime ([here](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/history/HistoryItem))

Seems like all the values are milliseconds since epoch so sorting these should not be difficult either.

I do have some questions about how we would implement the session search. If we were to allow searching through sessions, would we allow results for Window sessions? It seems the window object has a title but I wonder how we would do searching for a user. What does the window title get set to when a window is closed? Is it something that is obvious for a user to search for? Or would it be better to add the window.tabs array for closed windows to the search results?

**Comment by pureooze** [comment]
Small update, I found some time to try and implement this recently. I have managed to get the tabs and closed sessions merged into one list. Still need to update the colors and icons but leaving it now for easy distinction. I do wonder though, would it be useful for users if we showed visual distinction between tabs, sessions and history?

![image](https://user-images.githubusercontent.com/2337524/40950531-286e74b6-6841-11e8-8faf-408c297ca756.png)
